ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -267,13 +267,6 @@ implTryAddTxs mpEnv = go []
         join $ atomically $ readTVar mpEnvStateVar >>= tryAdd
       where
         tryAdd is
-          -- This transaction already exists within the mempool.
-          -- Note that we don't treat this as an error/rejection case.
-          | implSnapshotHasTx is (txId firstTx)
-          = return $ go
-              ((firstTx, MempoolTxAdded):acc)
```

Comment:
It seems that I originally made a mistake here actually. This should have been `MempoolTxAlreadyInMempool` rather than `MempoolTxAdded`.

I also need to remove the `MempoolTxAlreadyInMempool` constructor of `MempoolAddTxResult` as part of this PR since we don't use it.